### PR TITLE
Add: MeshCutter wasm. 

### PR DESCRIPTION
**Pull request by gonnavis**
Follow & wasm version of https://github.com/webaverse/metaverse-utils/pull/1

Related PR: https://github.com/webaverse/app-wasm/pull/19 , https://github.com/webaverse/mesh-cut-example/pull/1

First usable version MeshCutter wasm.

Need many improvement afterwards, such as:

1. Not hard-code cut `plane`.
1. Use `struct` to make codes less verbose.
1. Do not hard code output alloc length. Try to pass back wasm output length to js.

### CODE REVIEW

File: physx.js [hunk 1]
```diff
@@ -704,6 +704,94 @@ const physxWorker = (() => {
...
+    const numOutFacesTypedArray = allocator.alloc(Uint32Array, 2);
+
+    // output lenght may bigger than input length?
+    // todo: how to pass unknonwn length output from wasm to js?
+    // see https://rob-blackbourn.github.io/blog/webassembly/wasm/array/arrays/javascript/c/2020/06/07/wasm-arrays.html #Passing output arrays to wasm
+    // whether this tut has solution?
+    const outPositionsTypedArray = allocator.alloc(Float32Array, numPositions * 10); // todo: do not hard code alloc length.
+    const outNormalsTypedArray = allocator.alloc(Float32Array, numNormals * 10); // todo: do not hard code alloc length.
+    const outUvsTypedArray = allocator.alloc(Float32Array, numUvs * 10); // todo: do not hard code alloc length.
+    const outFacesTypedArray = allocator.alloc(Uint32Array, numFaces * 10); // todo: do not hard code alloc length.
```
Comment (avaer) [inline physx.js:757]: This is good use of the `Allocator` but we need to `free()` it when we are done or we leak memory from C++.

**Review by avaer — CHANGES_REQUESTED** [review]
Small change.

File: physx.js (hunk 1 above)
Comment (avaer) [reply physx.js:757]: https://github.com/webaverse/app/blob/aa2905e229fd04695742b5de8534c07e1f707571/physx.js#L871

Comment (gonnavis) [reply physx.js]: Thanks! Added https://github.com/webaverse/app/pull/2211/commits/c011b3e4757c1e2119d776146a5aba8e57d41d7c .

File: physx.js [hunk 2]
```diff
@@ -704,6 +704,89 @@ const physxWorker = (() => {
...
+    const numOutUvsTypedArray = moduleInstance.HEAPF32.subarray(head, tail);
+    head = tail;
+    tail = head + (numOutPositionsTypedArray[0] + numOutPositionsTypedArray[1]);
+    const outPositions = moduleInstance.HEAPF32.subarray(head, tail);
+    head = tail;
+    tail = head + (numOutNormalsTypedArray[0] + numOutNormalsTypedArray[1]);
+    const outNormals = moduleInstance.HEAPF32.subarray(head, tail);
+    head = tail;
+    tail = head + (numOutUvsTypedArray[0] + numOutUvsTypedArray[1]);
+    const outUvs = moduleInstance.HEAPF32.subarray(head, tail);
```
Comment (avaer) [inline physx.js:778]: This looks like illegal aliasing to me.

File: physx.js [hunk 3]
```diff
@@ -704,6 +704,89 @@ const physxWorker = (() => {
...
+      uvsTypedArray.byteOffset,
+      numUvs,
+      facesTypedArray.byteOffset,
+      numFaces,
+
+      positionTypedArray.byteOffset,
+      quaternionTypedArray.byteOffset,
+      scaleTypedArray.byteOffset,
+    );
+    allocator.freeAll();
```
Comment (avaer) [inline physx.js:759]: Looks like this output memory is never freed? It just leaks?

File: physx.js (hunk 2 above)
Comment (avaer) [reply physx.js]: https://github.com/webaverse/app/pull/2211/commits/7f0c3443645d425e8153b22d42f6080f4db0e0bb

File: physx.js (hunk 3 above)
Comment (avaer) [reply physx.js:759]: https://github.com/webaverse/app/pull/2211/commits/7f0c3443645d425e8153b22d42f6080f4db0e0bb

**Comment by avaer** [comment]
Massive conflict due to whitespace change.

**Pull request merged**